core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/delete/DeleteKeystoresHandler.java

```diff
@@ -0,0 +1,229 @@
+    } catch (final IllegalArgumentException | JsonProcessingException e) {
+      handleInvalidRequest(context, e);
+      return;
+    }
+
+    // normalize incoming keys to delete
+    final List<String> pubkeysToDelete =
+        parsedBody.getPubkeys().stream()
+            .map(IdentifierUtils::normaliseIdentifier)
+            .collect(Collectors.toList());
+
+    // load active keys
+    final Set<String> activePubkeys =
+        signerProvider.availableIdentifiers().stream()
+            .map(IdentifierUtils::normaliseIdentifier)
+            .collect(Collectors.toSet());
+
+    final List<DeleteKeystoreResult> results = new ArrayList<>();
+    final List<String> keysToExport = new ArrayList<>();
+    // process each incoming key individually
+    for (String pubkey : pubkeysToDelete) {
+      try {
+        final boolean isActive = activePubkeys.contains(pubkey);
+        final Optional<Path> keystoreConfigFile = findKeystoreConfigFile(pubkey);
+
+        // check that key is active
+        if (!isActive) {
+          // if not active, check if we ever had this key registered in the slashing DB
+          final boolean wasRegistered =
+              slashingProtection
+                  .map(protection -> protection.isRegisteredValidator(Bytes.fromHexString(pubkey)))
+                  .orElse(false);
+          // if it was registered previously, return not_active and add to list of keys to export,
+          // otherwise not_found
+          if (wasRegistered) {
+            keysToExport.add(pubkey);
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_ACTIVE, ""));
+          } else {
+            results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.NOT_FOUND, ""));
+          }
+          continue;
+        }
+        // Remove active key from memory first, will stop any further signing with this key
+        signerProvider.removeSigner(pubkey).get();
+        // Then, delete the corresponding keystore file
+        if (keystoreConfigFile.isPresent()) {
+          Files.delete(keystoreConfigFile.get());
+        }
+        // finally, add result response
+        keysToExport.add(pubkey);
+        results.add(new DeleteKeystoreResult(DeleteKeystoreStatus.DELETED, ""));
+      } catch (Exception e) {
+        results.add(
+            new DeleteKeystoreResult(
+                DeleteKeystoreStatus.ERROR, "Error deleting keystore file: " + e.getMessage()));
+      }
+    }
+
+    // export slashing protection data for 'deleted' and 'not_active' keys
+    String slashingProtectionExport = null;
+    if (slashingProtection.isPresent()) {
+      try {
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        slashingProtection.get().exportWithFilter(outputStream, keysToExport);
+        slashingProtectionExport = outputStream.toString(StandardCharsets.UTF_8);
+      } catch (Exception e) {
+        LOG.debug("Failed to export slashing data", e);
```

Comment:
Can test be added for the case that slashing protection data can't be exported and errors need to be returned for the keys

Reply:
how do you simulate an export failure? any examples in other tests?